testing_utils_test.go

```diff
@@ -142,6 +145,24 @@ func newFakeMarathonEndpoint(t *testing.T, config *Config) *endpoint {
 	}
 }
 
+// basicAuthMiddleware handles basic auth
+func basicAuthMiddleware(username, password string, next http.HandlerFunc) func(http.ResponseWriter, *http.Request) {
+	unauthorized := "{\"message\": \"invalid username or password\"}"
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		u, p, found := r.BasicAuth()
+		if !found && (username != "" || password != "") {
+			http.Error(w, unauthorized, 401)
+			return
+		}
+		if username != u || password != p {
+			http.Error(w, unauthorized, 401)
+			return
+		}
```

Comment:
I had to run through the various possible code paths a couple of times in order to convince myself that this is probably correct. This makes me think that the middleware might be complex enough to deserve a test of its own.

FWIW, at my organization if we need to write a test for a test helper located in a `*_test.go` file, we put it in a `*_test_test.go` file. Looks a bit weird but actually follows the naming logic quite well from my perspective.